doc/diagnostics/CallingConventions.md

```diff
@@ -0,0 +1,159 @@
+<!--
+Copyright (c) 2020, 2020 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at http://eclipse.org/legal/epl-2.0
+or the Apache License, Version 2.0 which accompanies this distribution
+and is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following Secondary
+Licenses when the conditions for such availability set forth in the
+Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+version 2 with the GNU Classpath Exception [1] and GNU General Public
+License, version 2 with the OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# x86
+
+## x86-64
+
+|Register|Linux System Linkage|Windows System Linkage|
+|--|--|--|
+|RAX|1st int return value ++|return value ++|1st int argument; int return value|
+|RBX|base pointer (optional) **|**|
+|RCX|4th int argument ++|1st argument ++|
+|RDX|3rd int argument; 2nd int return value ++|2nd argument ++|
+|RSP|stack pointer **|stack pointer **|
```

Comment:
Can we try to see how the tables will look by using emojis instead of "++" and "**". To me the best thing to do would be to use soft background colors on the table cells to indicate preserved vs. volatile registers. However unfortunately GitHub markdown doesn't support colors at all, not even in hacked together ways. So alternatives:

1. Render the table with colors manually and upload in this markdown in image form
2. Try to use colored emojis

The first option would certainly look better, but it is not searchable or copyable since it is an image. Second option is just pure markdown but not as easy to read. Here is the table for x86:

|Register|Linux System Linkage|Windows System Linkage|
|--|--|--|
|RAX|:large_blue_circle: 1st int return value |:large_blue_circle: return value |1st int argument; int return value|
|RBX|:red_circle: base pointer (optional) |:red_circle: |
|RCX|:large_blue_circle: 4th int argument |:large_blue_circle: 1st argument |
|RDX|:large_blue_circle: 3rd int argument; 2nd int return value |:large_blue_circle: 2nd argument |
|RSP|:red_circle: stack pointer |:red_circle: stack pointer |
|RBP|:red_circle: frame pointer (optional) |:red_circle: frame pointer (optional) |
|RSI|:large_blue_circle: 2nd int argument |:red_circle: |
|RDI|:large_blue_circle: 1st int argument |:red_circle: |
|R8|:large_blue_circle: 5th int argument |:large_blue_circle: 3rd argument |
|R9|:large_blue_circle: 6th int argument |:large_blue_circle: 4th argument |
|R10|:large_blue_circle: |:large_blue_circle: |
|R11|:large_blue_circle: |:large_blue_circle: |
|R12|:red_circle: |:red_circle: |
|R13|:red_circle: |:red_circle: |
|R14|:red_circle: |:red_circle: |
|R15|:red_circle: |:red_circle: |
|XMM0|:large_blue_circle: 1st float argument; 1st float return value |:large_blue_circle: 1st argument; return value |
|XMM1|:large_blue_circle: 2nd float argument; 2nd float return value |:large_blue_circle: 2nd argument |
|XMM2|:large_blue_circle: 3rd float argument |:large_blue_circle: 3rd argument |
|XMM3|:large_blue_circle: 4th float argument |:large_blue_circle: 4th argument |
|XMM4|:large_blue_circle: 5th float argument |:large_blue_circle: |
|XMM5|:large_blue_circle: 6th float argument |:large_blue_circle: |
|XMM6|:large_blue_circle: 7th float argument |:red_circle: |
|XMM7|:large_blue_circle: 8th float argument |:red_circle: |
|XMM8|:large_blue_circle: |:red_circle: |
|XMM9|:large_blue_circle: |:red_circle: |
|XMM10|:large_blue_circle: |:red_circle: |
|XMM11|:large_blue_circle: |:red_circle: |
|XMM12|:large_blue_circle: |:red_circle: |
|XMM13|:large_blue_circle: |:red_circle: |
|XMM14|:large_blue_circle: |:red_circle: |
|XMM15|:large_blue_circle: |:red_circle: |

:large_blue_circle: - caller saved
:red_circle: - callee saved

Reply:
I think it looks reasonable enough not to warrant using an image. What do you think?

Reply:
Yeah I don't think an image is a good idea. I think it looks decent the way it is. The other alternative is using raw HTML https://github.github.com/gfm/#html-blocks.

Reply:
Perhaps adding a separate unnamed column for the emojis may improve readibility. I've tried that here:

|Register||Linux System Linkage||Windows System Linkage|
|--|--|--|--|--|
|RAX|:large_blue_circle:| 1st int return value |:large_blue_circle:| return value |1st int argument; int return value|
|RBX|:red_circle:| base pointer (optional) |:red_circle:| |
|RCX|:large_blue_circle:| 4th int argument |:large_blue_circle:| 1st argument |
|RDX|:large_blue_circle:| 3rd int argument; 2nd int return value |:large_blue_circle:| 2nd argument |
|RSP|:red_circle:| stack pointer |:red_circle:| stack pointer |
|RBP|:red_circle:| frame pointer (optional) |:red_circle:| frame pointer (optional) |
|RSI|:large_blue_circle:| 2nd int argument |:red_circle:| |
|RDI|:large_blue_circle:| 1st int argument |:red_circle:| |
|R8|:large_blue_circle:| 5th int argument |:large_blue_circle:| 3rd argument |
|R9|:large_blue_circle:| 6th int argument |:large_blue_circle:| 4th argument |
|R10|:large_blue_circle:| |:large_blue_circle:| |
|R11|:large_blue_circle:| |:large_blue_circle:| |
|R12|:red_circle:| |:red_circle:| |
|R13|:red_circle:| |:red_circle:| |
|R14|:red_circle:| |:red_circle:| |
|R15|:red_circle:| |:red_circle:| |
|XMM0|:large_blue_circle:| 1st float argument; 1st float return value |:large_blue_circle:| 1st argument; return value |
|XMM1|:large_blue_circle:| 2nd float argument; 2nd float return value |:large_blue_circle:| 2nd argument |
|XMM2|:large_blue_circle:| 3rd float argument |:large_blue_circle:| 3rd argument |
|XMM3|:large_blue_circle:| 4th float argument |:large_blue_circle:| 4th argument |
|XMM4|:large_blue_circle:| 5th float argument |:large_blue_circle:| |
|XMM5|:large_blue_circle:| 6th float argument |:large_blue_circle:| |
|XMM6|:large_blue_circle:| 7th float argument |:red_circle:| |
|XMM7|:large_blue_circle:| 8th float argument |:red_circle:| |
|XMM8|:large_blue_circle:| |:red_circle:| |
|XMM9|:large_blue_circle:| |:red_circle:| |
|XMM10|:large_blue_circle:| |:red_circle:| |
|XMM11|:large_blue_circle:| |:red_circle:| |
|XMM12|:large_blue_circle:| |:red_circle:| |
|XMM13|:large_blue_circle:| |:red_circle:| |
|XMM14|:large_blue_circle:| |:red_circle:| |
|XMM15|:large_blue_circle:| |:red_circle:| |

Reply:
Alternatively we can get rid of the red circles and just leave the blue ones since the decision is binary; a register is volatile if and only if it is not preserved. We can then then just assume all empty cells are volatile. Not sure if that is clearer. It may make the table look less busy though. I'd like to hear thoughts of others.

Reply:
I'd be in favour having filled and open circles or something since if you are colour blind the above is really hard to grok. Could also use a letter like R = caller, E = callee or something...

Reply:
:large_blue_circle: and :white_circle: maybe? I couldn't find a hollow circle. Or maybe :black_square_button: and :white_square_button:?

Reply:
How about heavy check mark, like following? 
Register | Caller Preserved | Linux System Linkage |   | Windows System Linkage
-- | -- | -- | -- | --
RAX |  :heavy_check_mark: | 1st int return value |  :heavy_check_mark: | return value
RBX |   | base pointer (optional) |   |  
RCX |  :heavy_check_mark: | 4th int argument |  :heavy_check_mark: | 1st argument
RDX |  :heavy_check_mark: | 3rd int argument; 2nd int return value |  :heavy_check_mark: | 2nd argument
RSP |   | stack pointer |   | stack pointer
RBP |   | frame pointer (optional) |   | frame pointer (optional)
RSI |  :heavy_check_mark: | 2nd int argument |   |  
RDI |  :heavy_check_mark: | 1st int argument |   |  
R8 |  :heavy_check_mark: | 5th int argument |  :heavy_check_mark: | 3rd argument
R9 |  :heavy_check_mark: | 6th int argument |  :heavy_check_mark: | 4th argument
R10 |  :heavy_check_mark: |   |  :heavy_check_mark: |  
R11 |  :heavy_check_mark: |   |  :heavy_check_mark: |  